components/core/src/package/install.rs

```diff
@@ -168,31 +168,29 @@ impl PackageInstall {
     /// sorted order).
     pub fn runtime_path(&self) -> Result<String> {
         let mut idents = HashSet::new();
-        let mut run_path = String::new();
-        for path in try!(self.paths()) {
-            run_path.push_str(&path.to_string_lossy());
-            idents.insert(self.ident().clone());
-        }
+        let mut run_paths: Vec<PathBuf> = Vec::new();
+
+        let mut p = try!(self.paths());
+        run_paths.append(&mut p);
+        idents.insert(self.ident().clone());
         let deps: Vec<PackageInstall> = try!(self.load_deps());
         for dep in deps.iter() {
-            for path in try!(dep.paths()) {
-                run_path.push(':');
-                run_path.push_str(&path.to_string_lossy());
-                idents.insert(dep.ident().clone());
-            }
+            let mut p = try!(dep.paths());
+            run_paths.append(&mut p);
+            idents.insert(dep.ident().clone());
         }
         let tdeps: Vec<PackageInstall> = try!(self.load_tdeps());
         for dep in tdeps.iter() {
             if idents.contains(dep.ident()) {
                 continue;
             }
-            for path in try!(dep.paths()) {
-                run_path.push(':');
-                run_path.push_str(&path.to_string_lossy());
-                idents.insert(dep.ident().clone());
-            }
+            let mut p = try!(dep.paths());
+            run_paths.append(&mut p);
+            idents.insert(dep.ident().clone());
         }
-        Ok(run_path)
+
+        let p = env::join_paths(&run_paths).expect("Failed to build path string");
```

Comment:
![](https://m.popkey.co/688f96/w6qWM.gif)


Thanks for making this platform independent.  Saves @mwrock and I a bit of work.

Plus making `hab pkg exec` work is a good thing too :)